### DISCUSSION TITLE
Include more informations in the map files itself

### DISCUSSION THREAD

**Discussion by IceflowRE** [redeclipse/base#1125] [Ideas and Feature Requests]
I think it would be a good idea too include
- map modes
- map mutators
- map size

in the map files itself and just scan at game start the files and extract the values from them or if a new map get recognized just get those values too then. This part needs some tweaks, however.
#### Pro:
- user maps have those values and no need to set them manually
- easier to maintain
#### Cons:
- maybe a higher RAM usage, because every map has now those values
- longer start up time, to scan them (add a setting not to scan the user directory?)

**Comment by shirepirate** [comment]
i dont agree that it is easier to maintain. having lists as variables is better for backwards compatibility and useful if you want to play a mode or mut that the map was not originally intended for.

**Comment by IceflowRE** [comment]
The lists wont be lost.
It will just be generated from reading the map files itself.
Nothing other will change.

**Comment by qreeves** [comment]
Modes and mutators are a subjective thing, so they're best left to the server side configuration. It doesn't need to be included in the map, so requiring it is just an unnecessary step. Map size is stored in the map, but I'm more guessing you mean number of players, for which there are two variables, `numplayers` being the optimal number of players for the map (with the bots filling spaces to meet this "minimum") and `maxplayers` which limits the maximum number of people who can be alive at once (a thing that was once trialled, but is now only accessible by changing a server side variable).

**Comment by IceflowRE** [comment]
> they're best left to the server side configuration

You wont say that if you have to find out and maintain which modes and mutators are best for a map and you have around ~400 playable user maps.

With map size i mean the vars `sv_smallmaps`, `sv_mediummaps` and `sv_largemaps`.

**Comment by IceflowRE** [comment]
> Modes and mutators are a subjective thing

Yes, let the creator of the maps decide!
(You can still vote another mode and mutator..)
